crates/uv/src/commands/project/format.rs

```
@@ -40,8 +40,17 @@ pub(crate) async fn format(
 
     let workspace_cache = WorkspaceCache::default();
     let project =
-        VirtualProject::discover(project_dir, &DiscoveryOptions::default(), &workspace_cache)
-            .await?;
+        VirtualProject::discover(project_dir, &DiscoveryOptions::default(), &workspace_cache).await;
+    let working_dir = match project {
+        // If we found a project, we use the project root
+        Ok(proj) => proj.root().to_owned(),
+        // If there is a problem finding a project, we just use the provided directory
+        // e.g. unmanaged projects
+        Err(WorkspaceError::MissingPyprojectToml)
+        | Err(WorkspaceError::MissingProject(_))
+        | Err(WorkspaceError::NonWorkspace(_)) => project_dir.to_owned(),
+        Err(err) => return Err(err.into()),
```

I'm not sure on the behaviour in this case. Should we propagate the error to the caller (previous behaviour), or in this case, should we `warn_user!`  about the error and fallback to `project_dir` (as the previous branch does)?

I prefer to not do "magic fallbacks" when there is an error such as this (for example, malformed pyproject.toml) and to just propagate them to the user instead of continue with the command. Let me know your thoughts please.

In https://github.com/astral-sh/uv/blob/main/crates/uv/src/commands/venv.rs#L97 such fallback is done, logging a warning in these cases and continue working using defaults